cmd/bootkube/render.go

```diff
@@ -55,6 +59,9 @@ func init() {
 	cmdRender.Flags().StringVar(&renderOpts.assetDir, "asset-dir", "", "Output path for rendered assets")
 	cmdRender.Flags().StringVar(&renderOpts.caCertificatePath, "ca-certificate-path", "", "Path to an existing PEM encoded CA. If provided, TLS assets will be generated using this certificate authority.")
 	cmdRender.Flags().StringVar(&renderOpts.caPrivateKeyPath, "ca-private-key-path", "", "Path to an existing Certificate Authority RSA private key. Required if --ca-certificate is set.")
+	cmdRender.Flags().StringVar(&renderOpts.etcdCAPath, "etcd-ca-path", "", "Path to an existing PEM encoded CA that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-certificate-path and --etcd-private-key-path, and must have etcd configured to use TLS with matching secrets.")
+	cmdRender.Flags().StringVar(&renderOpts.etcdCertificatePath, "etcd-certificate-path", "", "Path to an existing certificate that will be used for TLS-enabled communication between the apiserver and etcd. Must be used in conjunction with --etcd-ca-path and --etcd-private-key-path, and must have etcd configured to use TLS with matching secrets.")
```

Comment:
This is definitely another mode of operation we could support, either in addition to or instead of the other modes.

I guess my thought was that if you were bringing your own etcd cluster you'd have the CA and certs for that already. The CA could certainly be the same one as what is used to generate the kubelet certs.

So, in total we have the following possible modes of operation:

1. user provides etcd ca, certs
2. user provides etcd ca, we use it to generate etcd certs
3. user provides kubelet ca, we use it to generate etcd certs
4. user provides no ca, we generate ca (shared for both kubelet and etcd) & use it to generate etcd certs

right now we support 1, 3, and 4. we could trivially add support for 2. what you describe would be to remove support for 1, and not support 2.

I'm cool with whatever you think makes sense :)